src/libraries/System.Net.Http/src/System/Net/Http/Headers/UriHeaderParser.cs

```diff
@@ -59,44 +60,28 @@ public override bool TryParseValue([NotNullWhen(true)] string? value, object? st
         // If not, just return the input value.
         internal static string DecodeUtf8FromString(string input)
         {
-            if (string.IsNullOrWhiteSpace(input))
+            if (!string.IsNullOrWhiteSpace(input))
             {
-                return input;
-            }
-
-            bool possibleUtf8 = false;
-            for (int i = 0; i < input.Length; i++)
-            {
-                if (input[i] > (char)255)
-                {
-                    return input; // This couldn't have come from the wire, someone assigned it directly.
-                }
-                else if (input[i] > (char)127)
+                int possibleUtf8Pos = input.AsSpan().IndexOfAnyExceptInRange((char)0, (char)127);
+                if (possibleUtf8Pos >= 0 &&
+                    input.AsSpan(possibleUtf8Pos).IndexOfAnyExceptInRange((char)0, (char)255) < 0)
                 {
-                    possibleUtf8 = true;
-                    break;
-                }
-            }
-            if (possibleUtf8)
-            {
-                byte[] rawBytes = new byte[input.Length];
-                for (int i = 0; i < input.Length; i++)
-                {
-                    if (input[i] > (char)255)
+                    Span<byte> rawBytes = input.Length <= 256 ? stackalloc byte[input.Length] : new byte[input.Length];
```

Comment:
Should we use const size for the `stackalloc` here like we typically do in CoreLib? Is `Systen.Net.Http` using `[SkipLocalsInit]` for the entire assembly?

```suggestion
                    Span<byte> rawBytes = input.Length <= 256 ? stackalloc byte[256] : new byte[input.Length];
```

Reply:
It would need to then also be sliced.  I don't think it's worth it.  This whole code path is suspect, anyway, and should likely be revisited.